### PR TITLE
Introduce BuildState.isProjectsCreated()

### PR DESCRIPTION
**Pull request by abstratt**
Follow-up to #25214.

Issue: #24341

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/build/BuildState.java [hunk 1]
```diff
@@ -69,6 +69,11 @@ public interface BuildState {
      */
     boolean isProjectsLoaded();
 
+    /**
+     * Is the mutable model for projects available?
+     */
+    boolean isProjectsAvailable();
```
Comment (alllex) [inline subprojects/core/src/main/java/org/gradle/internal/build/BuildState.java:75]: 🤔 The name seems to be too generic, and reading the javadoc is required to understand the meaning.
How about `isProjectsMutable()` or `isProjectsAvailableForMutation()` ?
An alternativ to keep the naming consistent with `ProjectState.isCreated()` and go with `isProjectsCreated()`

Comment (bamboo) [reply subprojects/core/src/main/java/org/gradle/internal/build/BuildState.java:75]: I'd go with `isProjectsCreated` to keep it consistent with `ProjectState#isCreated` and even include a `@see` link.

**Review by bamboo — APPROVED** [review]

File: subprojects/core/src/main/java/org/gradle/internal/build/BuildState.java (hunk 1 above)
Comment (abstratt) [reply subprojects/core/src/main/java/org/gradle/internal/build/BuildState.java]: 👍

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Pull request merged**